### DISCUSSION TITLE
how to signin directly after signup.?

### DISCUSSION THREAD

**Discussion by mohibullahjam** [nextauthjs/next-auth#11903] [Help]
is there a way to sign in the user directly after the signup completion in v5? i am using a verification code for signup and the requirement is to automatically sign in the user the code verification is successful is there a way to achieve that.? I am tried using the sign in from next-auth right after the code is verified but it returns error for invalid credentials. but when i go back to the login page and use the same credentials it sign ins perfectly. Can anyone help with that.?

**Comment by ezeparziale** [comment]
I think you need to use the signIn callback in the auth.config.ts
